Assets/Scripts/Game/Formulas/FormulaHelper.cs

```diff
@@ -958,60 +960,66 @@ public static int CalculateBackstabDamage(int damage, int backstabbingLevel)
+                var enemyTarget = target as EnemyEntity;
+                if (enemyTarget.MobileEnemy.Affinity == MobileAffinity.Human)
                 {
-                    damage += attacker.Level;
+                    if (((int)attacker.Career.HumanoidAttackModifier & (int)DFCareer.AttackModifier.Bonus) != 0)
+                        damage += attacker.Level;
+                    if (((int)attacker.Career.HumanoidAttackModifier & (int)DFCareer.AttackModifier.Phobia) != 0)
+                        damage -= attacker.Level;
                 }
-                if (((int)attacker.Career.DaedraAttackModifier & (int)DFCareer.AttackModifier.Phobia) != 0)
+                else if (enemyTarget.GetEnemyGroup() == DFCareer.EnemyGroups.Undead)
                 {
-                    damage -= attacker.Level;
+                    if (((int)attacker.Career.UndeadAttackModifier & (int)DFCareer.AttackModifier.Bonus) != 0)
+                        damage += attacker.Level;
+                    if (((int)attacker.Career.UndeadAttackModifier & (int)DFCareer.AttackModifier.Phobia) != 0)
+                        damage -= attacker.Level;
                 }
-            }
-            else if (AITarget.GetEnemyGroup() == DFCareer.EnemyGroups.Humanoid)
-            {
-                if (((int)attacker.Career.HumanoidAttackModifier & (int)DFCareer.AttackModifier.Bonus) != 0)
+                else if (enemyTarget.GetEnemyGroup() == DFCareer.EnemyGroups.Daedra)
                 {
-                    damage += attacker.Level;
+                    if (((int)attacker.Career.DaedraAttackModifier & (int)DFCareer.AttackModifier.Bonus) != 0)
+                        damage += attacker.Level;
+                    if (((int)attacker.Career.DaedraAttackModifier & (int)DFCareer.AttackModifier.Phobia) != 0)
+                        damage -= attacker.Level;
                 }
-                if (((int)attacker.Career.HumanoidAttackModifier & (int)DFCareer.AttackModifier.Phobia) != 0)
+                else if (enemyTarget.GetEnemyGroup() == DFCareer.EnemyGroups.Animals)
                 {
-                    damage -= attacker.Level;
+                    if (((int)attacker.Career.AnimalsAttackModifier & (int)DFCareer.AttackModifier.Bonus) != 0)
+                        damage += attacker.Level;
+                    if (((int)attacker.Career.AnimalsAttackModifier & (int)DFCareer.AttackModifier.Phobia) != 0)
+                        damage -= attacker.Level;
                 }
             }
-            else if (AITarget.GetEnemyGroup() == DFCareer.EnemyGroups.Animals)
+            else // Assume target is player if not an enemy
             {
-                if (((int)attacker.Career.AnimalsAttackModifier & (int)DFCareer.AttackModifier.Bonus) != 0)
+                if ((target as PlayerEntity).RaceTemplate.Name == TextManager.Instance.GetLocalizedText("vampire")) // Vampires are undead, therefore add undead modifier
```

Comment:
Better to use `GameManager.Instance.PlayerEffectManager.HasVampirism()` here than match by string.

Reply:
Ah, I didn't notice that method. Much better.